python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -26,3 +39,144 @@ class BooleanOps(DataTypeOps):
     @property
     def pretty_name(self) -> str:
         return 'booleans'
+
+    def add(self, left, right) -> Union["Series", "Index"]:
+        if not is_valid_operand_for_numeric_arithmetic(right, allow_bool=False):
+            raise TypeError(
+                "Addition can not be applied to %s and the given type." % self.pretty_name)
+
+        if isinstance(right, numbers.Number) and not isinstance(right, bool):
+            left = left.spark.transform(lambda scol: scol.cast(as_spark_type(type(right))))
+            return left + right
+        else:
+            assert isinstance(right, IndexOpsMixin)
```

Comment:
I guess `right` of type `spark.Column` could fall into this path.

Reply:
I see what you mean. I don't find an existing code path for `right` to be `spark.Column` for `boolean_ops` though.

Reply:
You mentioned that `spark.Column` is allowed to be operands (https://github.com/apache/spark/pull/32611#discussion_r636607352), and if so, we should handle it properly.

> And mypy likes this. Would you prefer to use cast instead?
What do you mean?

Reply:
Now `spark.Column` is disallowed in `is_valid_operand_for_numeric_arithmetic`.

Reply:
Shall we disable `spark.Column` as operands for other Ops class in a separate PR?